src/Nocarrier/Hal.php

```diff
@@ -240,7 +240,7 @@ public function setData(Array $data = null)
     public function getData($key = null)
     {
         if ($key) {
-            return isset($this->data[$key]) ? $this->data[$key] : null;
+            return isset($this->data[$key]) ? $this->data[$key] : array();
```

Comment:
I think the concern issue here was that until now getData has only _ever_ returned an array (and the docblock still states that's all is can return). It's not a BC break though because people would not have passed a parameter into getData before and that's what triggers it to return a scalar.

If you could update the docbloc to @return mixed that would cover it.